File: sdk/samples/all in on code/Visual Studio 2010/VBAzureBingMaps/WebRole1/WindowsLiveProfile.vb
'***************************** Module Header ******************************\
'* Module Name:	WindowsLiveProfile.vb
'* Project:		AzureBingMaps
'* Copyright (c) Microsoft Corporation.
'* 
'* A class corresponds to the response
'* from Windows Live Messenger Connect profile API.
'* Used by WCF syndication API.
'* 
'* This source is subject to the Microsoft Public License.
'* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
'* All other rights reserved.
'* 
'* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
'* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
'* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'\**************************************************************************


Imports System.Collections.Generic
Imports System.Runtime.Serialization

''' <summary>
''' A class corresponds to the response
''' from Windows Live Messenger Connect profile API.
''' Used by WCF syndication API.
''' </summary>
<DataContract([Namespace]:="http://schemas.microsoft.com/ado/2007/08/dataservices")> _
Public Class WindowsLiveProfile
    <DataMember()> _
    Public Property Addresses As List(Of element)
    <DataMember()> _
    Public Property BirthMonth As Integer
    <DataMember()> _
    Public Property Emails As List(Of element)
    <DataMember()> _
    Public Property FirstName As String
    <DataMember()> _
    Public Property Gender As String
    <DataMember()> _
    Public Property LastName As String
    <DataMember()> _
    Public Property Location As String
    <DataMember()> _
    Public Property PhoneNumbers As List(Of element)
End Class

<DataContract([Namespace]:="http://schemas.microsoft.com/ado/2007/08/dataservices")> _
Public Class element
    <DataMember()> _
    Public Property Address As String
    <DataMember()> _
    Public Property Type As String
    <DataMember()> _
    Public Property City As String
    <DataMember()> _
    Public Property CountryRegion As String
End Class

File: sdk/samples/all in on code/Visual Studio 2010/CSAzureBingMaps/WebRole1/WindowsLiveProfile.cs
/****************************** Module Header ******************************\
* Module Name:	WindowsLiveProfile.cs
* Project:		AzureBingMaps
* Copyright (c) Microsoft Corporation.
* 
* A class corresponds to the response
* from Windows Live Messenger Connect profile API.
* Used by WCF syndication API.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System.Collections.Generic;
using System.Runtime.Serialization;

namespace AzureBingMaps.WebRole
{
    /// <summary>
    /// A class corresponds to the response
    /// from Windows Live Messenger Connect profile API.
    /// Used by WCF syndication API.
    /// </summary>
    [DataContract(Namespace = "http://schemas.microsoft.com/ado/2007/08/dataservices")]
    public class WindowsLiveProfile
    {
        [DataMember]
        public List<element> Addresses { get; set; }
        [DataMember]
        public int BirthMonth { get; set; }
        [DataMember]
        public List<element> Emails { get; set; }
        [DataMember]
        public string FirstName { get; set; }
        [DataMember]
        public string Gender { get; set; }
        [DataMember]
        public string LastName { get; set; }
        [DataMember]
        public string Location { get; set; }
        [DataMember]
        public List<element> PhoneNumbers { get; set; }
    }

    [DataContract(Namespace = "http://schemas.microsoft.com/ado/2007/08/dataservices")]
    public class element
    {
        [DataMember]
        public string Address { get; set; }
        [DataMember]
        public string Type { get; set; }
        [DataMember]
        public string City { get; set; }
        [DataMember]
        public string CountryRegion { get; set; }
    }
}